torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -469,6 +469,64 @@ std::exception_ptr ProcessGroupNCCL::checkForNCCLErrorsInternal(
   return nullptr;
 }
 
+#ifdef ENABLE_NCCL_P2P_SUPPORT
+ncclResult_t ProcessGroupNCCL::ncclAlltoall(
+    void* sendbuff,
+    void* recvbuff,
+    size_t count,
+    size_t elem_size,
+    ncclDataType_t type,
+    ncclComm_t comm,
+    cudaStream_t stream) {
+  int numRanks;
+  size_t rank_diff = count * elem_size;
+  C10D_NCCL_CHECK(ncclCommCount(comm, &numRanks));
+  C10D_NCCL_CHECK(ncclGroupStart());
+  for (int r = 0; r < numRanks; r++) {
+    C10D_NCCL_CHECK(ncclSend(
+        ((char*)sendbuff) + r * rank_diff, count, type, r, comm, stream));
+    C10D_NCCL_CHECK(ncclRecv(
+        ((char*)recvbuff) + r * rank_diff, count, type, r, comm, stream));
+  }
+  C10D_NCCL_CHECK(ncclGroupEnd());
+  return ncclSuccess;
+}
+
+ncclResult_t ProcessGroupNCCL::ncclAlltoallv(
```

Comment:
Looks like these two are only used in this file. I wonder if it makes sense to put them in an anonymous namespace in the cpp file and not mention them in the hpp file as private methods?

Reply:
Fixed.